server/controllers/payment.js

```diff
@@ -0,0 +1,190 @@
+const User = require("../models/User");
+const StripeServices = require("../services/StripePayment");
+const asyncHandler = require("express-async-handler");
+const {
+  stripePayment,
+  createCustomer,
+  getCustomer,
+  subscriptionOneTimePayment,
+  createIntent,
+  getPaymentMethods,
+} = StripeServices;
+
+// @route POST /:id/single-pay
+// @desc create single payment checkout session
+// @access Private
+exports.payment = asyncHandler(async (req, res, next) => {
+  const id = req.user.id;
+  if (id === req.params.id) {
+    await stripePayment(req, res, function (err) {
+      if (err) {
+        return res.json({
+          success: false,
+          errors: {
+            title: "Stripe payment failed",
+            detail: err.message,
+            error: err,
+          },
+        });
+      }
+    });
+
+    res
+      .status(200)
+      .send({ message: "The payment session has been successfully created", url: req.url });
+  }
+});
+
+// @route GET /:id/create-customer
+// @desc create customer
+// @access Private
+exports.createCustomer = asyncHandler(async (req, res, next) => {
+  const id = req.user.id;
+  const user = await User.findById(id);
+  req.email = user.email;
+
+  if (id === req.params.id) {
+    await createCustomer(req, res, function (err) {
+      if (err) {
+        return res.json({
+          success: false,
+          errors: {
+            title: "Create customer failed",
+            detail: err.message,
+            error: err,
+          },
+        });
+      }
+    });
+
+    res
+      .status(200)
+      .send({ message: "The customer has been successfully created", customer: req.createdCustomer });
+  }
+});
+
+// @route GET /:id/new-customer-create-intent
+// @desc create customer and intent
+// @access Private
+exports.newCustomerCreateIntent = asyncHandler(async (req, res, next) => {
+  const id = req.user.id;
+  const user = await User.findById(id);
+  req.email = user.email;
+
+  if (id === req.params.id) {
+    await createCustomer(req, res, function (err) {
+      if (err) {
+        return res.json({
+          success: false,
+          errors: {
+            title: "Create customer failed",
+            detail: err.message,
+            error: err,
+          },
+        });
+      }
+    });
```

Comment:
Decorating your function with `async` keyword allows you to use `await` and assign the return value of a Promise to a variable instead of passing a callback.

Reply:
The function returns the variable in req property and there is error handler in the server. So, I remove the callback.